components/remote_settings/src/cache.rs

```diff
@@ -0,0 +1,167 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::RemoteSettingsResponse;
+use std::collections::HashSet;
+
+/// Interface for caching remote settings data
+///
+/// This is implemented by the remote settings consumer, sometimes with foreign code via UniFFI.
+///
+/// This interface has no error handling.  Consumers should report any errors themselves.  If there
+/// is an error in `get()` simply return `None` and the remote settings client will re-fetch any
+/// data.
+pub trait RemoteSettingsCache: Send + Sync {
+    /// Update the cached value
+    fn store(&self, value: String);
```

Comment:
Note: Does iOS need to provide a path for the cache or is this something that is already handled by remote settings?

Reply:
My thought was that iOS would implement this interface and pass that to the remote settings client.  Behind the scenes you could use whatever backing store made sense for the data.

Reply:
Yup, thats what my plan is. I got confused by the naming but what I will do is have a quick chat/call with you when this gets merged. Can show you the caching and if we are missing an API or need to make changes we can do that.